dashboard/test/controllers/pd/workshop_daily_survey_controller_test.rb

```diff
@@ -1029,6 +1029,77 @@ class WorkshopDailySurveyControllerTest < ActionDispatch::IntegrationTest
       assert_select 'h1', text: 'Thank you for submitting today’s survey.'
     end
 
+    test 'AYW1 pre-survey link shows foorm survey' do
+      setup_academic_year_workshop
+      sign_in @enrolled_academic_year_teacher
+
+      pre_survey_links = %w(/pd/AYW1/pre /pd/AYW1/pre/module/1 /pd/AYW1/pre/module/2)
+      pre_survey_links.each do |link|
+        get link
+        assert_response :success
+        assert_template :new_general_foorm
+      end
+    end
+
+    test 'AYW1 post-survey link shows foorm survey for attended teacher' do
+      setup_academic_year_workshop
+      sign_in @enrolled_academic_year_teacher
+      create :pd_attendance,
+        session: @academic_year_workshop.sessions[0],
+        teacher: @enrolled_academic_year_teacher,
+        enrollment: @academic_year_enrollment
+
+      pre_survey_links = %w(/pd/AYW1/post/in_person /pd/AYW1/post/module/1 /pd/AYW1/post/module/2 /pd/AYW1/pre/module/1_2)
+      pre_survey_links.each do |link|
+        get link
+        assert_response :success
+        assert_template :new_general_foorm
+      end
+    end
+
+    test 'AYW1 post-survey link shows no attendance for teacher with no attendance' do
+      setup_academic_year_workshop
+      sign_in @enrolled_academic_year_teacher
+
+      pre_survey_links = %w(/pd/AYW1/post/in_person /pd/AYW1/post/module/1 /pd/AYW1/post/module/2 /pd/AYW1/post/module/1_2)
+      pre_survey_links.each do |link|
+        get link
+        assert_response :success
+        assert_no_attendance
+      end
+    end
+
+    test 'AYW1_2 pre-survey link shows foorm survey' do
+      setup_two_day_academic_year_workshop
+      sign_in @enrolled_two_day_academic_year_teacher
+
+      get '/pd/AYW1_2/pre'
+      assert_response :success
+      assert_template :new_general_foorm
+    end
+
+    test 'AYW1_2 post-survey link shows foorm survey for attended teacher' do
+      setup_two_day_academic_year_workshop
+      sign_in @enrolled_two_day_academic_year_teacher
+
+      create :pd_attendance,
+        session: @two_day_academic_year_workshop.sessions[0],
+        teacher: @enrolled_two_day_academic_year_teacher,
+        enrollment: @two_day_academic_year_enrollment
+      get '/pd/AYW1_2/post/in_person'
+      assert_response :success
+      assert_template :new_general_foorm
+    end
+
+    test 'invalid AYW survey links show appropriate errors' do
```

Comment:
Test name seems like a misnomer. Maybe `'workshop enrollee can only load appropriate survey links` that both checks they can't load survey(s) they shouldn't be able to load, and can load the survey they should be able to load?

Reply:
Or could make the test actually check when invalid AYW survey links are used (eg, `/AYW/this_is_an_invalid_link`

Reply:
yeah I will update, I originally planned to include more links and didn't update the title